android/src/main/kotlin/com/f2fk/geofence_foreground_service/GeofenceForegroundServicePlugin.kt

```diff
@@ -68,134 +67,86 @@ class GeofenceForegroundServicePlugin : FlutterPlugin, MethodCallHandler, Activi
-
-                        channel.description = "A channel for receiving geofencing notifications"
-
-                        val notificationManager =
-                            activity?.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
-
-                        notificationManager.createNotificationChannel(channel)
-                    }
-
-                    ContextCompat.startForegroundService(context, serviceIntent)
-                    result.success(true)
-                } catch (e: Exception) {
-                    result.success(false)
-                }
-            }
-
-            "stopGeofencingService" -> {
-                try {
-                    context.stopService(serviceIntent)
-                    result.success(true)
-                } catch (e: Exception) {
-                    result.success(false)
-                }
-            }
+    override fun onMethodCall(call: MethodCall, result: Result) = when (call.method) {
+        ApiMethods.startGeofencingService -> startGeofencingService(call, result)
+        ApiMethods.stopGeofencingService -> stopGeofencingService(result)
+        ApiMethods.isForegroundServiceRunning -> result.success(context.isServiceRunning(GeofenceForegroundService::class.java))
+        ApiMethods.addGeofence -> addGeofence(Zone.fromJson(argumentsMap(call.arguments)), result)
+        ApiMethods.addGeoFences -> addGeoFences(ZonesList.fromJson(argumentsMap(call.arguments)), result)
+        ApiMethods.removeGeofence -> removeGeofence(listOf(call.argument(Constants.zoneId)!!), result)
+        else -> result.notImplemented()
+    }
 
-            "isForegroundServiceRunning" -> {
-                result.success(context.isServiceRunning(GeofenceForegroundService::class.java))
-            }
+    private fun startGeofencingService(call: MethodCall, result: Result) {
+        try {
+            SharedPreferenceHelper.saveCallbackDispatcherHandleKey(
+                context,
+                call.argument<Long>(Constants.callbackHandle)!!
+            )
 
-            "addGeofence" -> {
-                val zone: Zone = Zone.fromJson(call.arguments as Map<String, Any>)
+            channelId = call.argument<String>(Constants.channelId)
+            contentTitle = call.argument<String>(Constants.contentTitle)
+            contentText = call.argument<String>(Constants.contentText)
+            serviceId = call.argument<Int>(Constants.serviceId)
+            isInDebugMode = call.argument<Boolean>(Constants.isInDebugMode) ?: false
 
-                addGeofence(zone, result)
+            val iconDataJson: Map<String, Any>? = call.argument<Map<String, Any>>(
+                Constants.iconData
+            )
+            if (iconDataJson != null) {
+                iconData = NotificationIconData.fromJson(iconDataJson)
             }
+            serviceIntent = serviceIntent(GeofenceServiceAction.SETUP.toString())
+            val channel = NotificationChannel(
```

Comment:
In the original code I was checking for the android version O and above, should we bring it back or is it not needed? I remember I was getting a warning to handle it that way